skills/csharp/experimental/genericitsmskill/Dialogs/CreateFlowURLDialog.cs

```diff
@@ -0,0 +1,147 @@
+    {
+        private readonly IConfiguration _config;
+        private readonly IDataProtectionProvider _dataProtectionProvider;
+        private readonly IStatePropertyAccessor<ActivityReferenceMap> _activityReferenceMapAccessor;
+        private readonly IStatePropertyAccessor<TicketIdCorrelationMap> _ticketIdCorrelationMapAccessor;
+        private readonly ConversationState _conversationState;
+
+        public CreateFlowURLDialog(
+            IServiceProvider serviceProvider)
+            : base(nameof(CreateFlowURLDialog), serviceProvider)
+        {
+            _config = serviceProvider.GetService<IConfiguration>();
+            _dataProtectionProvider = serviceProvider.GetService<IDataProtectionProvider>();
+            _conversationState = serviceProvider.GetService<ConversationState>();
+            _activityReferenceMapAccessor = _conversationState.CreateProperty<ActivityReferenceMap>(nameof(ActivityReferenceMap));
+            _ticketIdCorrelationMapAccessor = _conversationState.CreateProperty<TicketIdCorrelationMap>(nameof(TicketIdCorrelationMap));
+
+            var sample = new WaterfallStep[]
+            {
+                GetAuthTokenAsync,
+                AfterGetAuthTokenAsync,
+                GetFlowInputAsync,
+                CreateFlowUrlAsync,
+                EndAsync,
+            };
+
+            AddDialog(new WaterfallDialog(nameof(CreateFlowURLDialog), sample));
+            AddDialog(new TextPrompt(DialogIds.NamePrompt));
+
+            InitialDialogId = nameof(CreateFlowURLDialog);
+        }
+
+        private async Task<DialogTurnResult> GetFlowInputAsync(WaterfallStepContext sc, CancellationToken cancellationToken)
+        {
+            // NOTE: Uncomment the following lines to access LUIS result for this turn.
+            // var luisResult = stepContext.Context.TurnState.Get<LuisResult>(StateProperties.SkillLuisResult);
+            var attachmentCard = GetFlowUrlInput.GetTicketIdInput("123", "123");
+            var cardReply = MessageFactory.Attachment(attachmentCard);
+            ResourceResponse resourceResponse = await sc.Context.SendActivityAsync(cardReply);
+
+            ActivityReferenceMap activityReferenceMap = await _activityReferenceMapAccessor.GetAsync(
+                sc.Context,
+                () => new ActivityReferenceMap(),
+                cancellationToken)
+            .ConfigureAwait(false);
+
+            // Store Activity and Thread Id
+            activityReferenceMap[sc.Context.Activity.Conversation.Id] = new ActivityReference
+            {
+                ActivityId = resourceResponse.Id,
+                ThreadId = sc.Context.Activity.Conversation.Id,
+                ConversationReference = sc.Context.Activity.GetConversationReference()
+            };
+            await _activityReferenceMapAccessor.SetAsync(sc.Context, activityReferenceMap).ConfigureAwait(false);
+
+            // Save Conversation State
+            await _conversationState
```

Comment:
can we have a TODO to convert this to waterfall dialog to Task module to improve the end user UX?